openstack/containerinfra/v1/quotas/results.go

```diff
@@ -0,0 +1,57 @@
+package quotas
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// CreateResult is the response of a Create operations.
+type CreateResult struct {
+	commonResult
+}
+
+// Extract is a function that accepts a result and extracts a quota resource.
+func (r commonResult) Extract() (*Quotas, error) {
+	var s *Quotas
+	err := r.ExtractInto(&s)
+	return s, err
+}
+
+type Quotas struct {
+	Resource  string    `json:"resource"`
+	CreatedAt time.Time `json:"created_at"`
+	UpdatedAt time.Time `json:"updated_at"`
+	HardLimit int       `json:"hard_limit"`
+	ProjectID string    `json:"project_id"`
+	ID        string    `json:"-"`
```

Comment:
The API code defines this as an int: https://github.com/openstack/magnum/blob/master/magnum/api/controllers/v1/quota.py#L39

But I'm OK with using a string. IDs of finite types usually don't last long and are eventually replaced with UUIDs, so using a string here might future-proof this package.

Reply:
Yes, originally I coded as int, but ran into situation where json was returning string, thus the special handling.